p4-fusion/p4_api.cc

```diff
@@ -196,33 +188,114 @@ ChangesResult P4API::Changes(const std::string& path, const std::string& from, i
 	return result;
 }
 
-DescribeResult P4API::Describe(const std::string& cl)
+DescribeResult P4API::Describe(const int cl)
 {
 	MTR_SCOPE("P4", __func__);
+
 	return Run<DescribeResult>("describe", { "-s", // Omit the diffs
-	                                           cl });
+	                                           std::to_string(cl) });
 }
 
-FileLogResult P4API::FileLog(const std::string& changelist)
+FileLogResult P4API::FileLog(const int changelist)
 {
 	return Run<FileLogResult>("filelog", {
 	                                         "-c", // restrict output to a single changelist
-	                                         changelist,
+	                                         std::to_string(changelist),
 	                                         "-m1", // don't get the full history, just the first entry.
 	                                         "//..." // rather than require the path to be passed in, just list all files.
 	                                     });
 }
 
-PrintResult P4API::PrintFiles(const std::vector<std::string>& fileRevisions)
+PrintResult P4API::PrintFiles(const std::vector<std::string>& fileRevisions, const std::function<void()>& onStat, const std::function<void(const char*, int)>& onOutput)
 {
 	MTR_SCOPE("P4", __func__);
 
 	if (fileRevisions.empty())
 	{
-		return PrintResult();
+		return { onStat, onOutput };
+	}
+
+	std::string argsString;
+	for (const std::string& stringArg : fileRevisions)
+	{
+		argsString += " " + stringArg;
```

Comment:
todo: Had to copy-paste the RunEx method here to create a result with the callbacks passed in.

Reply:
fixed